test/python/apps/polkadot.py

```diff
@@ -0,0 +1,151 @@
+    CONFIRM = 0x01
+
+class SignP1:
+    INIT = 0x00
+    ADD = 0x01
+    LAST = 0x02
+
+class SignP2Last:
+    ED25519 = 0x00
+    SR25119 = 0x01
+
+DOT_CONF = create_currency_config("DOT", "Polkadot", ("DOT", 18))
+
+DOT_PACKED_DERIVATION_PATH = bitcoin_pack_derivation_path(BtcDerivationPathFormat.LEGACY, "m/44'/354'/0'/0'/0'")
+
+DOT_PACKED_DERIVATION_PATH_SIGN_INIT = bytes([0x2c, 0x00, 0x00, 0x80,
+                                              0x62, 0x01, 0x00, 0x80,
+                                              0x00, 0x00, 0x00, 0x80,
+                                              0x00, 0x00, 0x00, 0x80,
+                                              0x00, 0x00, 0x00, 0x80])
+
+
+class PolkadotClient:
+    CLA = 0x90
+    def __init__(self, client):
+        self._client = client
+
+    @property
+    def client(self):
+        return self._client
+
+    def get_pubkey(self):
+        msg = self.client.exchange(self.CLA, ins=Command.GET_ADDRESS, p1=0, p2=0, data=DOT_PACKED_DERIVATION_PATH_SIGN_INIT)
+        return msg.data[:32].hex().encode()
+
+    def sign_init(self):
+        return self.client.exchange(self.CLA, ins=Command.SIGN_TX, p1=SignP1.INIT, data=DOT_PACKED_DERIVATION_PATH_SIGN_INIT)
+
+    def sign_add(self,tx_chunk):
+        return self.client.exchange(self.CLA, ins=Command.SIGN_TX, p1=SignP1.ADD,data=tx_chunk)
+
+    def sign_last(self,tx_chunk):
+        return self.client.exchange(self.CLA, ins=Command.SIGN_TX, p1=SignP1.LAST, p2=SignP2Last.ED25519, data=tx_chunk)
+
+    def verify_signature(self,hex_key:bytes,signature:bytes,message:bytes) -> bool :
+        # Create a VerifyKey object from a hex serialized public key
+        verify_key = VerifyKey(hex_key, encoder=HexEncoder)
+        # Check the validity of a message's signature
+        try:
+            verify_key.verify(message, signature, encoder=HexEncoder)
+        except BadSignatureError:
+            print("Wrong signature.")
+            return False
+        except Exception as e :
+            print("Something went wrong.")
+            print(e)
+            print(traceback.format_exc())
+            return False
+        else:
+            print("Signature is ok.")
+            return True
+
+    def craft_valid_polkadot_transaction(address, send_amount, fees, memo) -> bytes:
+        return Method.BALANCE_TRANSFER.to_bytes(2, "big") \
+               + AccountIdLookupType.ID.to_bytes(1, "big") \
+               + _polkadot_address_to_pk(address) \
+               + _format_amount(send_amount) \
+               + UNKNOWN \
+               + SPEC_VERSION.to_bytes(4, "little") \
+               + TX_VERSION.to_bytes(4, "little") \
+               + GENESIS_HASH \
+               + BLOCK_HASH
+
+    def craft_invalid_polkadot_transaction(address, send_amount, fees, memo) -> bytes:
+        force_transfer = Method.BALANCE_FORCE_TRANSFER.to_bytes(2, "big") \
+                       + bytes([0x00, 0xdc, 0x5a, 0xda, 0x10, 0xee, 0xdd, 0x89, 0x81, 0x92,
```

Comment:
Could we reuse this invalid byte array in other tests ? If so, maybe put it in a variable

Reply:
This function will be called by every test doing a `wrong method`